lib/Backend/Func.cpp

```diff
@@ -2068,6 +2075,78 @@ Func::GetForInEnumeratorArrayOffset() const
         + this->m_forInLoopBaseDepth * sizeof(Js::ForInObjectEnumerator);
 }
 
+void
+Func::SetLazyBailOutThunkLabel(IR::LabelInstr *label)
+{
+    Assert(label != nullptr &&
+        label->m_opcode == Js::OpCode::LazyBailOutThunkLabel &&
+        this->lazyBailOutThunkLabel == nullptr &&
+        this->IsInPhase(Js::Phase::FinalLowerPhase));
+
+    this->lazyBailOutThunkLabel = label;
+}
+
+IR::LabelInstr *
+Func::GetLazyBailOutThunkLabel() const
+{
+    Assert(this->lazyBailOutThunkLabel != nullptr);
+    return this->lazyBailOutThunkLabel;
+}
+
+void
+Func::SetHasLazyBailOut()
+{
+    this->hasLazyBailOut = true;
+}
+
+bool
+Func::HasLazyBailOut() const
+{
+    AssertMsg(
+        this->isPostRegAlloc,
+        "We don't know whether a function has lazy bailout until after RegAlloc"
+    );
+    return this->hasLazyBailOut;
+}
+
+void
+Func::AllocateLazyBailOutRecordArgSlotIfNeeded()
```

Comment:
>AllocateLazyBailOutRecordArgSlotIfNeeded [](start = 6, length = 40)

The common naming pattern for these kinds of functions in Chakra is Ensure*. Consider renaming according to that.